js/media/fieldmanager-media.js

```diff
@@ -20,44 +64,158 @@ $( document ).on( 'click', '.fm-media-button', function( event ) {
 		fm_media_frame[ $el.attr('id') ].open();
 		return;
 	}
+	var selectedImages = [],
+		inputVal = $el.parent().find('.fm-media-id').val();
+
+	if ( inputVal.length ) {
+		selectedImages = inputVal.split(',');
+	}
+
+	// Modification to gallery editing workflow
+	var mediaFrame = wp.media.view.MediaFrame.Post.extend({
+
+		createStates: function() {
+			var options = this.options;
+
+			this.states.add([
+
+				new wp.media.controller.Library({
+					id:         'gallery',
+					title:      options.title,
+					priority:   40,
+					toolbar:    'main-gallery',
+					filterable: 'uploaded',
+					multiple:   'add',
+					editable:   false,
+
+					library:  wp.media.query( _.defaults({
+						type: 'image'
+					}, options.library ) )
+				}),
+
+				new wp.media.controller.GalleryEdit({
+					library: options.selection,
+					editing: options.editing,
+					menu:    'gallery'
+				}),
+
+				new wp.media.controller.GalleryAdd({
+				})
+
+			]);
+
+		}
+
+	});
+
+	var query_args = {
+		'type': 'image',
+		'post__in': selectedImages,
+		'orderby': 'post__in',
+		'perPage': -1
```

Comment:
Not sure how VIP is going to feel about this. Might need to add some transparent pagination.